### PR TITLE
Update iceberg to 1.7.0

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionData.java [hunk 1]
```diff
@@ -137,6 +137,7 @@ public static Object getValue(JsonNode partitionValue, Type type)
                 return partitionValue.asInt();
             case LONG:
             case TIMESTAMP:
+            case TIMESTAMP_NANO:
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionData.java:140]: nano will be mapped to long timestamp, so this won't be there.

keep as not implemented and add a todo + link to https://github.com/trinodb/trino/issues/19753

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java [hunk 2]
```diff
@@ -1062,7 +1062,7 @@ public void testCreateTableWithUnsupportedNestedFieldPartitioning()
                 "\\QUnable to parse partitioning value: Cannot partition by non-primitive source field: struct<3: child: optional string>");
         assertQueryFails(
                 "CREATE TABLE test_partitioned_table_nested_field_inside_array (parent ARRAY(ROW(child VARCHAR))) WITH (partitioning = ARRAY['\"parent.child\"'])",
-                "\\QPartitioning field [parent.child] cannot be contained in a array");
+                "\\QPartitioning field [parent.element.child] cannot be contained in a array");
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java:1065]: do you know why?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergFileOperations.java [hunk 3]
```diff
@@ -124,8 +124,6 @@ public void testCreateTable()
         assertFileSystemAccesses("CREATE TABLE test_create (id VARCHAR, age INT)",
                 ImmutableMultiset.<FileOperation>builder()
                         .add(new FileOperation(METADATA_JSON, "OutputFile.create"))
-                        .add(new FileOperation(SNAPSHOT, "InputFile.length"))
-                        .add(new FileOperation(SNAPSHOT, "InputFile.newStream"))
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergFileOperations.java:128]: oh dear, nice

**Review by findepi — APPROVED** [review]

**Comment by findinpath** [comment]
There's another PR already ongoing for this update - see https://github.com/trinodb/trino/pull/23986 cc @ebyhr

**Pull request closed without merging**